gamla/functional_generic.py

```diff
@@ -271,3 +271,17 @@ def case(predicates_and_mappers: Tuple[Tuple[Callable, Callable], ...]):
 
 
 case_dict = compose_left(dict.items, tuple, case)
+
+
+def apply_spec(spec):
+    if anymap(asyncio.iscoroutinefunction, spec.values()):
+
+        async def apply_spec_async(input_value):
+            promises = []
+            for k, v in spec.items():
```

Comment:
Why loop and not map on items?

Reply:
you're right, changed.

(the reason I did it this way, is because I would need to use `apply` to reverse the relationship between function and value, as the iterable is of functions rather than values)